src/simulation_lifecycle/simulation.c

```diff
@@ -197,92 +196,60 @@ int write_sim_config(const cJSON *simulation_config, char *config_json) {
 int run_sim(const cJSON *simulation_config){
 
     /* Get model name - e.g. 2_2_agent_sir_config */
-    cJSON *model = cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_MODEL_ID);
+    char *model_name = cJSON_GetStringValue(cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_MODEL_ID));
 
     /* Check that model ID is provided using a valid format. */
-    if (model == NULL || !cJSON_IsString(model)) {
+    if (model_name == NULL) {
         return SIM_MODEL_SELECTION_INVALID;
     }
 
     /* Check that the model exists. */
-    char *model_path = concat(SIM_MODEL_LIBRARY, cJSON_GetStringValue(model));
-    int model_found = executable_exists(model_path);
-    free(model_path);  // NOTE: concat uses the heap for allocating memory -> we need to free it to avoid memory leakage!
-    if (!model_found) {
+    char *model_path = concat(SIM_MODEL_LIBRARY, model_name);
+    if (!executable_exists(model_path)) {
         return SIM_MODEL_SELECTION_INVALID;
     }
 
-    char *model_name = cJSON_GetStringValue(model);
-
-    /* Get current directory. */
-    char current_dir[MAX_LEN];
-    getcwd(current_dir, sizeof(current_dir));
-
-    /* Change to model executable files directory. */
-    chdir(SIM_MODEL_LIBRARY);
-
     /* Get config output path which is relative to SIM_MODEL_LIBRARY. */
     char *config_path = cJSON_GetStringValue(cJSON_GetObjectItemCaseSensitive(simulation_config, SIM_CONFIG_OUTPUT_PATH));
 
     /* Check that config output path is provided using a valid format. */
     if (config_path == NULL) {
```

Comment:
It no longer checks if the path exists. If you write this: `if (config_path == NULL || !file_exists(config_path)) {` and if you change the `config_output_path`` in the required json file in test/data/run_simulation/ to 
`"config_output_path": "../test/data/run_simulation/config.json"`, all tests for run_sim will passed.